packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -251,13 +251,25 @@ Future<XcodeBuildResult> buildXcodeProject({
   );
 
   if (hasPlugins()) {
-    final String currentGeneratedXcconfig = readGeneratedXcconfig(app.appDirectory);
+    final String iosPath = fs.path.join(fs.currentDirectory.path, app.appDirectory);
+    // If the Xcode project, Podfile, or Generated.xcconfig have changed since
+    // last run, pods should be updated.
+    final Fingerprinter fingerprinter = new Fingerprinter(
+      fingerprintPath: fs.path.join(getIosBuildDirectory(), 'pod_inputs.fingerprint'),
+      paths: <String>[
+        _getPbxProjPath(app.appDirectory),
+        fs.path.join(iosPath, 'Podfile'),
+        fs.path.join(iosPath, 'Flutter', 'Generated.xcconfig'),
+      ],
+      properties: <String, String>{},
+    );
     await cocoaPods.processPods(
       appIosDirectory: appDirectory,
       iosEngineDir: flutterFrameworkDir(buildInfo.mode),
       isSwift: app.isSwift,
-      flutterPodChanged: previousGeneratedXcconfig != currentGeneratedXcconfig,
+      dependenciesChanged: !await fingerprinter.doesFingerprintMatch()
     );
+    await fingerprinter.writeFingerprint();
```

Comment:
Is it possible that the user doesn't even have pod installed, runs processPods, dumps an error message but goes through, writes a fingerprint. The user installs pods but it never runs again because the fingerprint matches?

Reply:
Maybe a question for @mravn-google, should we check for cocoapods being installed during inject plugins before it gets here? I don't have a strong opinion either way.

Reply:
I've added a boolean return value from `processPods` that indicates whether we actually ran pod install or not, and only write if so.